beanstalk_worker/worker_start_scan.py

```diff
@@ -26,212 +26,424 @@
+                )
+        elif scanner == "registry.centos.org/pipeline-images/scanner-rpm-verify":
+            # should receive the JSON data loaded
+            status, json_data = self.scanner_rpmverify_object.run(image_under_test)
+            if not status:
+                logger.log(
+                    level=logging.WARNING,
+                    msg="Failed to run the scanner %s" % scanner
+                )
+            else:
+                logger.log(
+                    level=logging.INFO,
+                    msg="Finished running %s scanner." % scanner)
+
+        # if scanner failed to run, this will return {}, do check at receiver end
+        return json_data
+
+    def run(self):
+        """"
+        Runs the listed atomic scanners on image under test
+
+        #FIXME: at the moment this menthod is returning the results of multiple
+        scanners in one json and sends over the bus
+        """
+        logger.log(level=logging.INFO, msg="Received job : %s" % job_info)
+
+        # TODO: Figure out why random tag (with date) is coming
+        # image_under_test = ":".join(self.job_info.get("name").split(":")[:-1])
+        image_under_test = self.job_info.get("name")
+        logger.log(level=logging.INFO, msg="Image under test:%s" % image_under_test)
+
+        # copy the job info into scanners data, as we are going to add logs and msg
+        scanners_data = self.job_info
+        scanners_data["msg"] = {}
+        scanners_data["logs"] = {}
+
+        # pull the image first, if failed move on to start_delivery
+        if not self.pull_image_under_test(image_under_test):
+            logger.log(
+                level=logging.INFO,
+                msg="Image pulled failed, moving job to delivery phase."
+            )
+            scanners_data["action"] = "start_delivery"
+            return False, scanners_data
+
+        # run the multiple scanners on image under test
+        for scanner in SCANNERS.keys():
+            data_temp = self.run_a_scanner(scanner, image_under_test)
+            if not data_temp:
+                continue
+            scanners_data["msg"][data_temp["scanner_name"]] = data_temp["msg"]
+            scanners_data["logs"][data_temp["scanner_name"]] = data_temp["logs"]
+
+        scanners_data["action"] = "notify_user"
+        # This field is needed for email worker to hint that this is scan
+        # result email and need to move to next phase of delivery
+        scanners_data["scan_results"] = True
+
+        # after all scanners are ran, remove the image
+        logger.log(
+                level=logging.INFO,
+                msg="Removing the image: %s" % image_under_test)
+
+        conn.remove_image(image=image_under_test, force=True)
+
+        # TODO: Check here if at least one scanner ran successfully
+        logger.log(level=logging.INFO, msg="Finished executing all scanners.")
+        return True, scanners_data
+
+
+class ScannerRPMVerify(object):
```

Comment:
There are two types of scanners -
 - scanners needing image_under_test to be mounted (readwrite)
 - scanners not needing image_under_test to be mounted (readonly)

I am planning to structure it in following way
1 - A main scanner handler (base class)
2 - A class for (all) readonly scanners ( inherit from 1)
3 -  A class for (all) readwrite scanners (inherit from 1)
N - Individual scanners data validations [..] (this will later be transformed into configs)


The reason, I did not put everything in single PR, is to minimize the total code addition / deletion. At the moment, we have 2 scanners, this PR is scoped around that.

Beside that I agree on the suggestion. (y)